definitions/checks/system_registration.rb

```diff
@@ -5,27 +5,32 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf')
+      system_self_registered?
     end
   end
 
   def run
-    if system_is_self_registerd?
-      warn! 'System is self registered'
-    else
-      puts 'System is not self registered'
-    end
+    warn! 'System is self registered'
   end
 
-  def system_is_self_registerd?
-    rhsm_hostname.casecmp(hostname).zero?
-  end
+  class << self
+    def rhsm_hostname
+      execute("grep '\\bhostname\\b' < #{rhsm_conf_file}  | grep -v '^#'").sub(/.*?=/, '').strip
+    end
 
-  def rhsm_hostname
-    execute("grep '\\bhostname\\b' < #{rhsm_conf_file}  | grep -v '^#'").sub(/.*?=/, '').strip
-  end
+    def rhsm_conf_file
+      '/etc/rhsm/rhsm.conf'
+    end
+
+    def rhsm_hostname_eql_hostname?
+      @result ||= rhsm_hostname.casecmp(hostname).zero?
+    end
 
-  def rhsm_conf_file
-    '/etc/rhsm/rhsm.conf'
+    def system_self_registered?
+      file_exists?('/etc/rhsm/rhsm.conf') &&
+        !feature(:foreman_server) &&
+        feature(:foreman_proxy).present? &&
```

Comment:
Similarly as the `:foreman_server` also the `:foreman_proxy` and `:katello` should rather use confine blocks than the `present?` method, to match with the other features

Reply:
Also, this condition should go to the confine block of this check

Reply:
You mean the whole condition of `system_self_registered` should belong to confine block? OR just
```
confine do
  file_exists?('/etc/rhsm/rhsm.conf') &&
    !feature(:foreman_server) &&
    feature(:foreman_proxy)
end
```